### ISSUE TITLE
Spec for a Symbol

### ISSUE THREAD

**Issue by mattbishop** [tbknl/specified#4]
I want to check that an object has a specific Symbol attached to it. How would I do this?

**Comment by tbknl** [comment]
Hi @mattbishop ,

Interesting question. For symbols I think that specified would need a new "type", something like `Type.symbol` to check that a specific type is of type "symbol". To check for a specific Symbol value you could use the generic `oneOf` constraint.

I will consider to include symbol types in the next release of specified. But until that time you could role your own type. The specified package is designed to be extensible, so you can easily create your own "types" and constraints, although in the current version there is not yet a stable interface to do that. So I suggest for now you write your type using the existing structure for types/specs. Here's an example of how that could look like:

```typescript
import { ValidationError, constrain, Constraint, verify } from "specified";

const Type_Symbol = {
    definition: {
        type: "symbol"
    },
    eval: (value: unknown) => {
        if (typeof value !== "symbol") {
            throw new ValidationError("Not a symbol.");
        }
        return value;
    }
};

const mySymbol = Symbol("symbolic");
const mySymbolSpec = constrain(Type_Symbol, [Constraint.generic.oneOf(mySymbol)]);
const symbolValue = verify(mySymbolSpec, mySymbol).value();
```

**Comment by mattbishop** [comment]
Thank you for the example, I will try it out. I don't suppose testing for a Symbol is very common so this solution may be just fine.

**Comment by tbknl** [comment]
`Type.symbol` is now included in version 0.4.0 of the specified package.
